### PR TITLE
add auth Wallet Import Format method (WIF)

### PR DESCRIPTION
**Pull request by Tortue95**

### CODE REVIEW

**Comment by M5oul** [comment]
Why there is so much authentication methods?
What is the advantage to use this one?

**Comment by Tortue95** [comment]
for papers wallet (that is the reason why i added the transaction on silkaj)

File: src/tools.py [hunk 1]
```diff
@@ -50,6 +50,36 @@ def get_seed_from_scrypt(salt, password, N=4096, r=16, p=1):
     seed = scrypt.hash(password, salt, N, r, p, 32)
     seedhex = nacl.encoding.HexEncoder.encode(seed).decode("utf-8")
     return seedhex
+    
+    
+def get_seed_from_wif(wif):
+    regex = re.compile('^[1-9A-HJ-NP-Za-km-z]*$')
+    if not re.search(regex, wif):
+        print("Error: the format of WIF is invalid")
+        exit()
```
Comment (jytou) [inline src/tools.py]: exit(1) to return the fact that there was an error

File: src/tools.py [hunk 2]
```diff
@@ -50,6 +50,36 @@ def get_seed_from_scrypt(salt, password, N=4096, r=16, p=1):
...
+
+    wif_bytes = b58_decode(wif)
+    if len(wif_bytes) != 35:
+        print("Error: the size of WIF is invalid")
+        exit()
```
Comment (jytou) [inline src/tools.py]: idem

File: src/tools.py [hunk 3]
```diff
@@ -50,6 +50,36 @@ def get_seed_from_scrypt(salt, password, N=4096, r=16, p=1):
...
+
+    checksum_from_wif = wif_bytes[-2:]
+    fi = wif_bytes[0:1]
+    seed = wif_bytes[1:-2]
+    seed_fi = wif_bytes[0:-2]
+
+    if fi != b'\x01':
+        print("Error: It's not a WIF format")
+        exit()
```
Comment (jytou) [inline src/tools.py]: idem

File: src/tools.py [hunk 4]
```diff
@@ -50,6 +50,36 @@ def get_seed_from_scrypt(salt, password, N=4096, r=16, p=1):
...
+
+    #checksum control 
+    checksum = nacl.hash.sha256(nacl.hash.sha256(seed_fi, nacl.encoding.RawEncoder),nacl.encoding.RawEncoder)[0:2]
+    if checksum_from_wif != checksum:
+        print("Error: bad checksum of the WIF")
+        exit()
```
Comment (jytou) [inline src/tools.py]: idem

File: src/tools.py [hunk 5]
```diff
@@ -50,6 +50,36 @@ def get_seed_from_scrypt(salt, password, N=4096, r=16, p=1):
...
+
+    seedhex = nacl.encoding.HexEncoder.encode(seed).decode("utf-8")
```
Comment (jytou) [inline src/tools.py]: I would simply return the encode call, but that's a personal preference. :)

File: src/tools.py (hunk 1 above)
Comment (Tortue95) [inline src/tools.py]: yes we need to patch it for all silkaj

Comment (jytou) [inline src/tools.py]: At least we can do them when we find them. I don't think there are so many. :)

**Comment by Tortue95** [comment]
Moul: We can remove the seed authentification if you want

**Comment by M5oul** [comment]
It's ok. I wanted to understand.
I didn't knew we could use many authentication methods to communicate with Duniter network.

**Comment by jytou** [comment]
Please do keep the seed authentication method. :)

File: src/tools.py (hunk 3 above)
Comment (Insoleet) [inline src/tools.py]: What would be perfect is too include the paper wallet verification directly in duniter-python-api so that Sakia could use it too ;)

**Pull request merged**